profile-googlefonts/src/license.rs

```diff
@@ -0,0 +1,77 @@
+use font_types::NameId;
+use fontspector_checkapi::prelude::*;
+use fontspector_checkapi::{testfont, FileTypeConvert};
+use read_fonts::TableProvider;
+use regex::Regex;
+use skrifa::MetadataProvider;
+
+// Although this is a /name/ check, it's really about licensing
+#[check(
+    id = "googlefonts/name/rfn",
+    title = "Name table strings must not contain the string 'Reserved Font Name'.",
+    rationale = "
+        Some designers adopt the \"Reserved Font Name\" clause of the OFL license. This
+        means that the original author reserves the rights to the family name and other
+        people can only distribute modified versions using a different family name.
+
+        Google Fonts published updates to the fonts in the collection in order to fix
+        issues and/or implement further improvements to the fonts. It is important to
+        keep the family name so that users of the webfonts can benefit from the updates.
+        Since it would forbid such usage scenario, all families in the GFonts collection
+        are required to not adopt the RFN clause.
+
+        This check ensures \"Reserved Font Name\" is not mentioned in the name table.
+    ",
+    proposal="https://github.com/fonttools/fontbakery/issues/1380",
+)]
+fn name_rfn(t: &Testable, _context: &Context) -> CheckFnResult {
+    let f = testfont!(t);
+    let mut problems = vec![];
+
+    let records = f.font().name()?.name_record();
+    let str_data = f.font().name()?.string_data();
+
+    for name in records {
+        let name_string = name.string(str_data)?
+                              .chars()
+                              .collect::<String>()
+                              .to_string();
+        if name_string.contains(
+            "This license is copied below, and is also available with a FAQ") {
+            /* This is the OFL text in a name table entry.
+               It contains the term 'Reserved Font Name' in one of its clauses,
+               so we will ignore this here. */
+            continue;
+        }
+
+        let familyname = f.font()
+                          .localized_strings(NameId::FAMILY_NAME)
+                          .english_or_first()
+                          .ok_or(CheckError::Error("No name ID 1".to_string()))?
+                          .chars()
+                          .collect::<String>();
+        let re = Regex::new(r"with [Rr]eserved [Ff]ont [Nn]ame '(?<rfn>[^']*)'").unwrap();
```

Comment:
I am trying to avoid using regular expressions in Rust as much as possible. They're quite different to Python. For starters, `Regex::new` actually parses the regex and creates a new regular expression engine at runtime,  which is very slow, and then executing the regex is also pretty slow. I used a regex for validating glyph names and it slowed things down massively; it turned out that 90% of fontspector's execution was spent constructing the regex engine and executing it. Caching the regex with a `lazy_static` was helpful but it was still worth rewriting it with simple string matching functions if possible.